### ISSUE TITLE
App crushes after unsuspending

### ISSUE THREAD

**Issue by JanczarKurek** [gergo-salyi/multibg-sway#1]
# Issue description
App crushes if system i unsuspended while it's running.

# Steps to reproduce
1. Run app
2. run `systemctl suspend`
3. Unsuspend

# Stacktrace
Running with `RUST_BACKTRACE=full` on debug build gives 
```
[2023-04-26T07:39:06Z DEBUG multibg_sway::wayland] New output, name: eDP-1, resolution: 1920x1080
[2023-04-26T07:39:06Z DEBUG multibg_sway::wayland] Using pixel format: Bgr888
[2023-04-26T07:39:07Z DEBUG multibg_sway::wayland] Loaded 2 wallpapers on new output for workspaces: _default, 4
[2023-04-26T07:39:07Z DEBUG multibg_sway::wayland] Background layer has been configured for output: eDP-1
[2023-04-26T07:39:07Z DEBUG multibg_sway::wayland] Setting wallpaper on output 'eDP-1' for workspace: 1
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 4, kind: Interrupted, message: "Interrupted system call" }', /home/janczarknurek/.cargo/registry/src/github.com-1ecc6299db9ec823/multibg-sway-0.1.1/src/main.rs:122:38
stack backtrace:
   0:     0x55b16dec50da - std::backtrace_rs::backtrace::libunwind::trace::hf11d87cc2d43e172
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x55b16dec50da - std::backtrace_rs::backtrace::trace_unsynchronized::h161efd06cff9ba70
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x55b16dec50da - std::sys_common::backtrace::_print_fmt::h3db77eb2c5fae718
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/sys_common/backtrace.rs:65:5
   3:     0x55b16dec50da - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h3ac2a42750789445
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x55b16deea06e - core::fmt::write::hfe53c6572a5f8908
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/fmt/mod.rs:1213:17
   5:     0x55b16dec23d5 - std::io::Write::write_fmt::h70632594b6222755
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/io/mod.rs:1682:15
   6:     0x55b16dec4ea5 - std::sys_common::backtrace::_print::hf387e79c0dfc7a4c
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/sys_common/backtrace.rs:47:5
   7:     0x55b16dec4ea5 - std::sys_common::backtrace::print::hc737969126dc91c4
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/sys_common/backtrace.rs:34:9
   8:     0x55b16dec668f - std::panicking::default_hook::{{closure}}::h6742236f814fdecf
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:267:22
   9:     0x55b16dec63cb - std::panicking::default_hook::h0266f560c6fab5bd
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:286:9
  10:     0x55b16dec6d99 - std::panicking::rust_panic_with_hook::ha0d0c9e604110127
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:688:13
  11:     0x55b16dec6b39 - std::panicking::begin_panic_handler::{{closure}}::h392fb174af40c06c
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:579:13
  12:     0x55b16dec558c - std::sys_common::backtrace::__rust_end_short_backtrace::haeac18b86ae1a9e3
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/sys_common/backtrace.rs:137:18
  13:     0x55b16dec6842 - rust_begin_unwind
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:575:5
  14:     0x55b16d621cf3 - core::panicking::panic_fmt::h87c8fa127debfca6
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/panicking.rs:64:14
  15:     0x55b16d6221a3 - core::result::unwrap_failed::h0a81b1af128cb591
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/result.rs:1790:5
  16:     0x55b16dbc1d0d - core::result::Result<T,E>::unwrap::h0d3803908b90538a
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/result.rs:1112:23
  17:     0x55b16d63e1b3 - multibg_sway::main::ha0531101c89e53fd
                               at /home/janczarknurek/.cargo/registry/src/github.com-1ecc6299db9ec823/multibg-sway-0.1.1/src/main.rs:122:9
  18:     0x55b16d680f7b - core::ops::function::FnOnce::call_once::hed3a4470c9f4822b
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/ops/function.rs:250:5
  19:     0x55b16d67948e - std::sys_common::backtrace::__rust_begin_short_backtrace::h134542d38e7c3a30
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/sys_common/backtrace.rs:121:18
  20:     0x55b16d654481 - std::rt::lang_start::{{closure}}::had99b746690fc32d
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/rt.rs:166:18
  21:     0x55b16debc8ec - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h5e6c4e00516dc27c
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/ops/function.rs:287:13
  22:     0x55b16debc8ec - std::panicking::try::do_call::hfc2c5ba951d3d673
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:483:40
  23:     0x55b16debc8ec - std::panicking::try::h31938cd7eb8e57c6
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:447:19
  24:     0x55b16debc8ec - std::panic::catch_unwind::h04cb5a122979d0dc
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panic.rs:140:14
  25:     0x55b16debc8ec - std::rt::lang_start_internal::{{closure}}::h8bc10d7812def12b
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/rt.rs:148:48
  26:     0x55b16debc8ec - std::panicking::try::do_call::hc26e7f80ee4901c8
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:483:40
  27:     0x55b16debc8ec - std::panicking::try::he69995efb0248735
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:447:19
  28:     0x55b16debc8ec - std::panic::catch_unwind::hcb2f9fd13dbc661e
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panic.rs:140:14
  29:     0x55b16debc8ec - std::rt::lang_start_internal::h794d1e8722b84ab9
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/rt.rs:148:20
  30:     0x55b16d65445a - std::rt::lang_start::h1fb483f1501a15c2
                               at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/rt.rs:165:17
  31:     0x55b16d63ed1e - main
  32:     0x7f1a4db61790 - <unknown>
  33:     0x7f1a4db6184a - __libc_start_main
  34:     0x55b16d622385 - _start
  35:                0x0 - <unknown>
```

**Comment by gergo-salyi** [comment]
Thanks!

I fixed it, version 0.1.2 is out now on crates.io and in AUR.

Please comment an OK if it works well now,
Thank You!

**Comment by JanczarKurek** [comment]
Thank you for a quick response, it's working now.
